### PR TITLE
Vector output fixes

### LINKED ISSUE #219: Detection boxes are placed incorrectly when using region filters wtih web services
There are a few issues with detection boxes that result after using a region filter.

1) A very small portion of the bounding box is clipped from the input bounding box

2) The CRS is never set for the vector output

### PR DESCRIPTION
**Pull request by avitebskiy**
- fixed output SpatialReference not being initialized in initMapServiceImage
- made region filter relative to the bbox to sync with the detection boxes
- added error checking for filter coordinate system mismatch

resolve #219

### CODE REVIEW

File: src/OpenSpaceNet.cpp [hunk 1]
```diff
@@ -308,13 +319,23 @@ void OpenSpaceNet::initFilter()
             for (const auto& filterFile : filterAction.second) {
                 FeatureSet filter(filterFile);
                 for (auto& layer : filter) {
-                    auto transform = TransformationChain { std::move(layer.spatialReference().to(imageSr)),
-                                                           std::move(image_->pixelToProj().inverse())};
+                    auto pixelToProj = dynamic_cast<const TransformationChain&>(*pixelToLL_);
```
Comment (rddesmond) [inline src/OpenSpaceNet.cpp]: `imageSr` is no longer used.. it's just obtained when it is needed on 328 below.  Please clean up the unused variable.

File: src/OpenSpaceNet.cpp [hunk 2]
```diff
@@ -325,7 +346,7 @@ void OpenSpaceNet::initFilter()
             } else if (action == "exclude") {
                 if (firstAction) {
                     OSN_LOG(info) << "User excluded regions first...automatically including the bounding box...";
-                    regionFilter_->add({{bbox_}});
+                    regionFilter_->add(Polygon(LinearRing(cv::Rect(0, 0, bbox_.width, bbox_.height))));
```
Comment (rddesmond) [inline src/OpenSpaceNet.cpp]: Can't you simplify this with
```
regionFilter_->add({{cv::Rect{0, 0, bbox_.width, bbox_.height}}});```

Comment (avitebskiy) [inline src/OpenSpaceNet.cpp]: No, that will break Mac build. Also this doesn't work. It's a GCC bug that this was allowed at all.

**Comment by avitebskiy** [comment]
test this please

**Comment by geovations** [comment]
test this please

**Pull request merged**